Typing in randomization.ts results in errors in using the linters

Functions from src/randomization.ts with arguments of type Array<TItem> are unhappy is some cases function:
```typescript
function randomizeStimuli<TItem extends ItemType> (
    original_stimuli : Array<TItem>,
    max_same_type: number = 2,
    type_key:string = 'item_type'
) : Array<TItem> | null {
    let constraints = {[type_key] : max_same_type}; // ES6 dependency.
    return randomizePrivate(original_stimuli, constraints, 0, 10);
}
```
Seems happy when you give it an array of objects in place

```typescript
randomizeStimuli([{item_type: "test"}, {item_type: "prac"});
```
However in the unit test, we have a function that generates an array e.g.
```typescript
let stimuli = createStimuli();
randomizeStimuli(stimuli);
```
, and then you'll get an error such as:

> tests/randomization.test.ts:108:13 - error TS2345: Argument of type 'object[]' is not assignable to parameter of type 'ItemType[]'.
>   Type 'object' is not assignable to type 'ItemType'.
>     Index signature for type 'string' is missing in type '{}'.
> 
> 108             stimuli,
>

--- a/src/randomization.ts
+++ b/src/randomization.ts
@@ -256,7 +256,7 @@ function randomizeStimuli<TItem extends ItemType> (
  */
 function randomizeStimuliConstraints<TItem extends ItemType> (
     original_stimuli: Array<TItem>,
-    constraints = {'item_type' : 2},
+    constraints: ConstraintMapping = {'item_type' : 2},
     max_tries = 10
 ) : Array<TItem> | null {
     return randomizePrivate(original_stimuli, constraints, 0, max_tries);

